### PR TITLE
[Feature][Connector-V2] Support IoTDB sink

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

Support IoTDB sink connector #1946

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/connector-v2/sink/IoTDB.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+### common options [string]
+
+Sink plugin common parameters, please refer to [Sink Common Options](common-options.md) for details
+
+## Examples
+
+```hocon
+sink {
+  Clickhouse {
```
Comment (ashulin) [inline docs/en/connector-v2/sink/IoTDB.md]:
```suggestion
  IoTDB {
```

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSink.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.seatunnel.connectors.seatunnel.iotdb.state.IoTDBSinkState;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.google.auto.service.AutoService;
+
+import java.io.IOException;
+
+@AutoService(SeaTunnelSink.class)
+public class IoTDBSink implements SeaTunnelSink<SeaTunnelRow, IoTDBSinkState, IoTDBCommitInfo, IoTDBAggregatedCommitInfo> {
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSink.java]:
```suggestion
public class IoTDBSink extend AbstractSimpleSink<SeaTunnelRow, Void> {
```
It's easier to implement，and you can delete actually useless classes;

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkWriter.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Optional;
+
+@Slf4j
+public class IoTDBSinkWriter implements SinkWriter<SeaTunnelRow, IoTDBCommitInfo, IoTDBSinkState> {
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkWriter.java]:
```suggestion
public class IoTDBSinkWriter extends AbstractSimpleSink<SeaTunnelRow, Void> {
```
Same as above.

**Comment by hailin0** [comment]
Disable iotdb spark-testcase due to conflict with libthrift package

spark-server/lib 2.4.3: libthrift-0.9.3
spark-hive 2.4.3: libthrift-0.9.3
iotdb-session 13.1: libthrift-0.14.1

**Comment by EricJoy2048** [comment]
> Disable iotdb spark-testcase due to conflict with libthrift package
> 
> spark-server/lib 2.4.3: libthrift-0.9.3 spark-hive 2.4.3: libthrift-0.9.3 iotdb-session 13.1: libthrift-0.14.1

It seems spark use `parent first` classload profile. So this connector can not support spark now. I suggest we add this notice to document and merge this pr.

**Comment by hailin0** [comment]
> > Disable iotdb spark-testcase due to conflict with libthrift package
> > spark-server/lib 2.4.3: libthrift-0.9.3 spark-hive 2.4.3: libthrift-0.9.3 iotdb-session 13.1: libthrift-0.14.1
> 
> It seems spark use `parent first` classload profile. So this connector can not support spark now. I suggest we add this notice to document and merge this pr.

updated

**Review by EricJoy2048 — COMMENTED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**